renku_notebooks/api/schemas.py

```diff
@@ -1,8 +1,45 @@
-from marshmallow import Schema, fields, post_load, post_dump
+from marshmallow import (
+    Schema,
+    fields,
+    post_load,
+    post_dump,
+    validates_schema,
+    ValidationError,
+)
 import collections
 
 from .. import config
-from .custom_fields import UnionField
+from .custom_fields import (
+    serverOptionCpuValue,
+    serverOptionMemoryValue,
+)
+from ..util.misc import read_server_options_file
+
+
+class LaunchNotebookRequestServerOptions(Schema):
+    defaultUrl = fields.String(required=True)
+    cpu_request = serverOptionCpuValue
+    mem_request = serverOptionMemoryValue
+    lfs_auto_fetch = fields.Bool(required=True)
+    gpu_request = fields.Integer(strict=True, validate=lambda x: x >= 0)
+
+    @validates_schema
+    def validate_server_options(self, data, **kwargs):
+        server_options = read_server_options_file()
+        for option in data.keys():
+            if server_options.get(option, {}).get("type", None) == "boolean":
```

Comment:
This does the right thing, but it took me a moment to process the logic. How about something like this?
``
if option not in server_options:
  continue
if server_options[option]["type"] == "boolean":
  continue
if data[option] not in server_options[option]["options"]:
  raise ValidationError...
 ```

Reply:
Ok sure I will keep the `get` statements to avoid errors when a key is missing.

Reply:
Don't server options always have a type and non-boolean server options always have options? Wouldn't in this case uncaught KeyError be more helpful in detecting a problem here than silently using an empty dictionary or list?

Reply:
I typed too soon. I think I can fully get away from the `get` statements with what you propose.

Reply:
@ableuler you are right. The `get` statements are gone.